### PR TITLE
Add Vendor domain to Claims processing application

### PR DESCRIPTION
**Pull request by dmahapatro**
*Description of changes:*
**Vendor Service (EKS implementation)**

Vendor service is a new domain that becomes a producer/consumer of the Insurance Claims processing EDA architecture as it has been elaborated in the [blog](https://aws.amazon.com/blogs/industries/building-a-modern-event-driven-application-for-insurance-claims-processing-part-2/). Below are the requirements that are met with this PR:

* EventBridge custom event bus has a new rule that evaluates for Vendor Service. 
* This rule is be provided by Vendor Service.
* The target of this rule is an SQS queue owned by Vendor Service.
* EKS deployment scales based on the events populated in SQS queue using [KEDA](https://keda.sh/docs/2.10/scalers/aws-sqs/).
* KEDA will scale EKS deployments based on number of messages in SQS queue (queue depth).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/services/vendor/infra/vendor-service.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  public readonly cluster: eks.Cluster;
+  public readonly vendorMetricsWidget: GraphWidget;
+
+  constructor(scope: Construct, id: string, props: VendorServiceProps) {
+    super(scope, id);
+
+    const region = Stack.of(this).region;
+
+    const clusterAdmin = new iam.Role(this, "AdminRole", {
+      assumedBy: new iam.AccountRootPrincipal(),
```
Comment (jihed) [inline lib/services/vendor/infra/vendor-service.ts:75]: You may need to use a dedicated IAM role here for deployment.

**Review by jihed — COMMENTED** [review]
Minot comment on IAM role. Otherwise, lgtm

File: lib/services/vendor/infra/vendor-service.ts (hunk 1 above)
Comment (dmahapatro) [reply lib/services/vendor/infra/vendor-service.ts:75]: @jihed Sorry, I did not understand. Isn't `clusterAdmin` a dedicated IAM role? What am I missing here?

**Review by jihed — APPROVED** [review]
lgtm

**Pull request merged**